testflinger_cli/__init__.py

```diff
@@ -468,15 +468,15 @@ def do_poll(self, job_id):
                 output = ""
                 try:
                     output = self.get_latest_output(job_id)
+                    if output:
+                        print(output, end="", flush=True)
+                    job_state = self.get_job_state(job_id)
+                    self.history.update(job_id, job_state)
```

Comment:
AIUI given the history is stored in a file and does not require a connection would it make sense to not bundle these together in a single try,except? 

Given you are also now catching IOError in get_job_state, I'm not sure exactly what these changes achieve

Reply:
We do still need to catch IOError here because of the chance that getting the current job state could trigger it. We can't just read it from the file, because we need the current state which may have changed. We just write it to the file to keep that current.

However, it's a fair question because we could wrap the entire section in a single try/except block and just handle the possible exceptions at the end of the whole thing, which is much simpler looking. I've changed it like that now, see what you think.